Tests/Foundation/Tests/TestDecimal.swift

```diff
@@ -860,8 +843,41 @@ class TestDecimal: XCTestCase {
     }
     
     func test_ULP() {
-        let x = 0.1 as Decimal
+        var x = 0.1 as Decimal
         XCTAssertFalse(x.ulp > x)
+
+        x = .nan
+        XCTAssertTrue(x.ulp.isNaN)
+        XCTAssertTrue(x.nextDown.isNaN)
+        XCTAssertTrue(x.nextUp.isNaN)
+
+        x = .greatestFiniteMagnitude
+        XCTAssertEqual(x.ulp, Decimal(string: "1e127")!)
+        XCTAssertEqual(x.nextDown, x - Decimal(string: "1e127")!)
+        XCTAssertTrue(x.nextUp.isNaN)
+
+        x = 1
+        XCTAssertEqual(x.ulp, Decimal(string: "1e-38")!)
+        XCTAssertEqual(x.nextDown, x - Decimal(string: "1e-38")!)
+        XCTAssertEqual(x.nextUp, x + Decimal(string: "1e-38")!)
+
+        x = .leastNonzeroMagnitude
+        // XCTAssertEqual(x.ulp, x)        // SR-6671
+        // XCTAssertEqual(x.nextDown, 0)   // SR-6671
+        // XCTAssertEqual(x.nextUp, x + x) // SR-6671
+        XCTAssertNotEqual(x.ulp, 0)
+        XCTAssertNotEqual(x.nextDown, x)
+        XCTAssertNotEqual(x.nextUp, x)
+
+        x = 0
+        XCTAssertEqual(x.ulp, Decimal(string: "1e-128")!)
+        XCTAssertEqual(x.nextDown, -Decimal(string: "1e-128")!)
+        XCTAssertEqual(x.nextUp, Decimal(string: "1e-128")!)
+
+        x = -1
+        XCTAssertEqual(x.ulp, (1 as Decimal).ulp)
+        XCTAssertEqual(x.nextDown, -((1 as Decimal).nextUp))
+        XCTAssertEqual(x.nextUp, -((1 as Decimal).nextDown))
```

Comment:
Could you add a few extra tests for these last 3 to test the result against absolute values eg:
`XCTAssertEqual(x.ulp, Decimal(string: "1e-38")!)`
in case there is an equal and opposite bug in both methods.